### PR TITLE
Button creation

### PR DESCRIPTION
**Pull request by niroz11**

### CODE REVIEW

File: src/Components/FilterControls.js [hunk 1]
```diff
@@ -21,32 +22,32 @@ class FilterControls extends Component {
   filterParades = () => {
```
Comment (raechelo) [inline src/Components/FilterControls.js:22]: here is our current function and how we're trying to accomplish filtering through all three options for seeing the parades.

`  filterParades = () => { 
    const { location, time, date } = this.state;
    const { parades } = this.props;
    let matchedIds = [];
    let matchedObjects = [];

    parades.forEach(parade => {
      if (!this.state.matchedIds.includes(parade.id) && 
      parade.location === location || 
      parade.dayOrNight === time ||
      parade.date === date)
      {
        matchedObjects.push(parade);
        matchedIds.push(parade.id);
      }
    });
    // this.setState({ matchedIds });
    console.log(matchedObjects)
    return matchedObjects
  }`

**Comment by raechelo** [comment]
We have a large portion of our functionality done, you can filter through the parades using the individual filters and we are now working at filtering using a combination of the three. We're having issues with getting it to work effectively and you can see our progress in the commit and the comments we have left in the code.

@brittanystoroz @thatpamiam @robbiejaeger

File: src/CardContainer.css [hunk 2]
```diff
@@ -1,14 +1,37 @@
...
+  color:white;
+  text-align: center;
+  margin-top: 130px;
+  margin-left: 65px;
+}
+
+.individual-card {
+  display: grid;
+  width: 300px;
 }
```
Comment (brittanydionigi) [inline src/CardContainer.css:37]: I would try to find a style for organizing and ordering your CSS properties. I particularly like the "group by type" approach described in this article: https://css-tricks.com/poll-results-how-do-you-order-your-css-properties/ Others do it alphabetically, which I like less, but it still offers some sort of predictable ordering that makes it easier for other developers to read and follow along with.

File: src/Components/CardContainer.js [hunk 3]
```diff
@@ -25,10 +24,11 @@ class CardContainer extends Component {
 }
 
     render() {
+      console.log(this.props.filteredParades)
       return(
           <div>
-           <section>
-            {this.displayLocationCards()}
+           <section className = "card-container">
+            {this.displayCards()}
```
Comment (brittanydionigi) [inline src/Components/CardContainer.js:31]: Nice job breaking this rendered portion of the JSX out into a method -- this is a common approach for dealing with large chunks of conditional rendering. Not entirely sure it's 100% necessary here, but you will see this pattern in many places in React apps.

File: src/Components/FilterControls.js [hunk 4]
```diff
@@ -1,86 +1,101 @@
...
     }
   }
 
-  getClickedLocation = (event) => {
-    let selectedLocation = event.target.value;
+  getClickedValue = (event) => {
     this.setState({
-      location: selectedLocation
-    }, this.filterByLocation)
+      [event.target.id]: event.target.value
```
Comment (brittanydionigi) [inline src/Components/FilterControls.js:18]: Confused by this setState method here...it looks like you're anticipating having a state property that represents `event.target.id` --- but in your component state you only have `location`, `time`, `date`, and `matchedIds` --- I'm not sure where or what state would be updated with this as written. 🤔

File: src/Components/FilterControls.js [hunk 5]
```diff
@@ -1,86 +1,101 @@
...
+    let matchedIds = [];
+    let matchedObjects = this.returnMatchingObjects();
+
+    parades.forEach(parade => {
+      if (!matchedObjects.includes(parade) && 
+      parade.location === location || 
+      parade.dayOrNight === time ||
+      parade.date === date)
+      {
+        matchedObjects.push(parade);
```
Comment (brittanydionigi) [inline src/Components/FilterControls.js:34]: What are you actually doing with this `matchedObjects` array? It looks like you create a variable for it, but then it's never actually used anywhere.

File: src/Components/FilterControls.js [hunk 6]
```diff
@@ -1,86 +1,101 @@
...
+  returnMatchingObjects = () => {
+    return this.props.parades.reduce((acc, parade) => {
+      this.state.matchedIds.forEach(id => {
+        if (parade.id === id) {
+          acc.push(parade)
+          console.log(parade)
+        }
     })
+      return acc
+    }, [])
```
Comment (brittanydionigi) [inline src/Components/FilterControls.js:50]: Might a `filter` work better here? If we filtered `this.props.parades` and our condition was instead:

`return this.state.matchedIds.includes(parade.id)` we should end up with a subset of the parades array that includes only parades that match an ID found in `this.state.matchedIds`

**Pull request merged**